src/modules/Game.class.js

```diff
@@ -23,22 +23,280 @@ class Game {
   constructor(initialState) {
     // eslint-disable-next-line no-console
     console.log(initialState);
+    this.isGameOver = false;
+    this.score = 0;
+    this.hasMovedOnce = false;
   }
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
+  moveLeft() {
+    if (this.isGameOver) return;
+    const prevState = this.getState();
+
+    const tbody = document.querySelector('tbody');
+    const rows = tbody.querySelectorAll('tr');
+
+    rows.forEach((row) => {
+      const columns = row.querySelectorAll('td');
+
+      const values = [];
+
+      columns.forEach((cell) => {
+        const value = cell.textContent.trim();
+
+        if (value !== '') {
+          values.push(value);
+        }
+      });
+
+      for (let i = 0; i < values.length; i++) {
+        if (values[i] === values[i + 1]) {
+          values[i] *= 2;
+          this.score += values[i];
+          values.splice(i + 1, 1);
+        }
+      }
+
+      columns.forEach((cell, i) => {
```

Comment:
You have similar logic in every method, try to put it in one helper function and use it everywhere.